### ISSUE TITLE
Use /await:strict to replace /await .

### ISSUE THREAD

**Issue by frg2089** [Baseflow/flutter-permission-handler#1045]
I think we should replace `/await` with `/await:strict`.
Because some features of `/await` are not supported in std c++ 20.

https://github.com/Baseflow/flutter-permission-handler/blob/2e47f05ae9dabee2bc671166301aa948259b39f2/permission_handler_windows/windows/CMakeLists.txt#L50

> The /await:strict option disables language extensions present in /await that weren't adopted into the C++20 standard. Use of such features results in a compiler error. The option also implements coroutine behaviors such as promise parameter preview. These behaviors aren't available under /await because of binary compatibility issues in older versions of Visual Studio.
[Microsoft Docs](https://learn.microsoft.com/cpp/build/reference/await-enable-coroutine-support?view=msvc-170#remarks)

**Comment by TimHoogstrate** [comment]
Dear @frg2089, This seems like a good idea. However, our team is lacking some knowledge about this subject. You can help us a lot by making a PR with the suggested changes. Next to this, do you know if this also works for older versions of c++?

Kind regards,

**Comment by TimHoogstrate** [comment]
It seems that we can do something like a ifdef c++20 for those who compile at c++ 20. (example)
```
#if __cplusplus >= 202002L
    // C++20 (and later) code
#endif
```

**Comment by mvanbeusekom** [comment]
Fixed per version 0.2.0 of the geolocator_windows plugin (referenced in geolocator version 10.0.0).
